src/CompiledContainer.php

```diff
@@ -120,4 +121,21 @@ protected function resolveFactory($callable, $entryName, array $extraParameters
             throw new InvalidDefinition("Entry \"$entryName\" cannot be resolved: " . $e->getMessage());
         }
     }
+
+    /**
+     * Resolve a placeholder in string definition
+     * - wrap possible NotFound exception to conform to the one from StringDefinition::resolveExpression.
+     */
+    protected function resolveStringPlaceholder($placeholder, $entryName)
```

Comment:
missing tests, doc blocks, I would put `@throws` there as well

Reply:
these tests cover this function indirectly: https://github.com/ovos/PHP-DI/blob/feature/compiler-optimize-string-definitions/tests/IntegrationTest/Definitions/StringDefinitionTest.php